src/util/play_time.rs

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    pub fn get_total_active_time(&self) -> Duration {
+        self.total_active_duration
+            + self
+                .current_session_start
+                .map_or(Duration::ZERO, |s| s.elapsed())
+    }
+
+    pub fn is_active(&self) -> bool {
+        self.current_session_start.is_some()
+    }
+
+    pub fn start_session(&mut self) {
+        if self.current_session_start.is_none() {
+            self.current_session_start = Some(Instant::now());
+            // Also update activity time immediately
+            self.last_activity_time = SystemTime::now();
+        }
+    }
+
+    /// Called periodically while recording is active to mark ongoing activity
+    pub fn tick_activity(&mut self) {
+        // Keep updating the "Last Seen" time to Now while playing.
+        // This ensures that when the recording stops, this timestamp
+        // marks the exact end of the session.
+        self.last_activity_time = SystemTime::now();
+    }
+
+    pub fn pause_session(&mut self) {
+        if let Some(start) = self.current_session_start.take() {
+            self.total_active_duration += start.elapsed();
+        }
+        // Ensure we mark the end of the session
+        self.last_activity_time = SystemTime::now();
+    }
+
+    /// Called ONLY when the Recorder stops specifically due to IDLE
+    pub fn cancel_idle_buffer(&mut self) {
+        // We subtract the idle duration (e.g. 30s) from the total
+        // because the Recorder includes it before triggering the stop.
+        let idle_buffer = constants::MAX_IDLE_DURATION;
+        if self.total_active_duration >= idle_buffer {
+            self.total_active_duration -= idle_buffer;
+        }
+    }
+
+    pub fn reset(&mut self) {
+        *self = Self::new();
+    }
+
+    pub fn should_reset(&self) -> bool {
+        let idle = self.last_activity_time.elapsed().unwrap_or_default();
+        let since_break = self.last_break_end.elapsed().unwrap_or_default();
+
+        idle > constants::PLAY_TIME_BREAK_THRESHOLD
+            || since_break > constants::PLAY_TIME_ROLLING_WINDOW
+    }
+
+    pub fn save(&self) -> Result<(), Box<dyn std::error::Error>> {
+        let path = get_persistent_dir()?.join(persistent::PLAY_TIME_STATE);
+        fs::write(&path, serde_json::to_string_pretty(self)?)?;
+        tracing::debug!("Saved play time state to {}", path.display());
+        Ok(())
+    }
+
+    pub fn load() -> Self {
+        let state = (|| -> Result<Self, Box<dyn std::error::Error>> {
+            let path = get_persistent_dir()?.join(persistent::PLAY_TIME_STATE);
+            Ok(serde_json::from_str(&fs::read_to_string(&path)?)?)
+        })()
+        .unwrap_or_else(|e| {
+            tracing::info!("Using default play time state: {}", e);
+            Self::new()
+        });
+
+        if state.should_reset() {
+            tracing::info!("Resetting play time state based on thresholds");
+            return Self::new();
```

Comment:
I suggest making the `state` `mut` and calling `state.reset()` here, so that we're working with the same `state` throughout this function